### PR TITLE
Add documentation for how to create an app

### LINKED ISSUE #452: Add documentation for create-app

### PR DESCRIPTION
**Pull request by marcuseide**
## Hey, I just made a Pull Request!

First part of `create-app` documentation.

Fixes https://github.com/spotify/backstage/issues/452

### CODE REVIEW

**Review by stefanalund — DISMISSED** [review]
This is an importan document. I think it would be good to put it in front of someone that is not already familiar with the project and see if it makes sense for them. Approving nonetheless.

File: docs/getting-started/create-a-plugin.md [hunk 1]
```diff
@@ -1,20 +1,28 @@
-# Create a plugin
+# Backstage Plugins
+
+A Backstage Plugin adds functionality to Backstage.
+
+## Create a plugin
 
 To create a new plugin, make sure you've run `yarn` and installed dependencies, then run the following on your command line (invoking the `backstage-cli`).
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply docs/getting-started/create-a-plugin.md]: Yeah I think it's best to be explicit about that. Imagining it's less confusing for people new to yarn.

File: docs/getting-started/create-an-app.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+│       ├── public
+│       │   └──  ...
+│       └── src
+│           ├── App.test.tsx
+│           ├── App.tsx
+│           ├── index.tsx
+│           ├── plugins.ts
+│           └── setupTests.ts
+└── plugins
+    └── welcome
```
Comment (stefanalund) [inline docs/getting-started/create-an-app.md:52]: Perhaps this should show "your-plugin-x" etc?

**Review by stefanalund — DISMISSED** [review]

**Comment by Rugvip** [comment]
We want to merge #465 before this

File: docs/getting-started/create-an-app.md (hunk 2 above)
Comment (Rugvip) [reply docs/getting-started/create-an-app.md:52]: This is what it generates right now

File: docs/getting-started/create-an-app.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Backstage App
+
+If you want to get set up quickly with your own Backstage project you can create a Backstage App.
```
Comment (Rugvip) [inline docs/getting-started/create-an-app.md:3]: "To get set up quickly with your own Backstage project ..." imo

File: docs/getting-started/create-an-app.md (hunk 2 above)
Comment (marcuseide) [reply docs/getting-started/create-an-app.md:52]: I'm thinking that this list of files and folders is what `create-app` provides for you when you run it. Then when you use `create-plugin` we have in our docs that it adds it to the `plugins/` directory.

File: docs/getting-started/create-an-app.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+A Backstage App is a monorepo setup with `lerna` that includes everything you need to run Backstage in your own environment.
+
+## Create an app
+
+If you have [`npx`](https://github.com/npm/npx) installed simply run:
```
Comment (Rugvip) [inline docs/getting-started/create-an-app.md:9]: npx comes bundled with npm/node, so no need to link to it. Could maybe reiterate that node v12 is needed

File: docs/getting-started/create-an-app.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+```bash
+npx @backstage/cli create-app
+```
+
+Or with `yarn`:
+
+```bash
+yarn add --dev @backstage/cli
+yarn run backstage-cli create-app
```
Comment (Rugvip) [inline docs/getting-started/create-an-app.md:19]: Not sure we want to recommend this. The directory structure you end up with doesn't make much sense imo. cwd/node_modules + cwd/package.json + cwd/app/node_modules + cwd/app/package.json + cwd/app/packages/app/package.json, etc

**Comment by guillaumebreton** [comment]
👋 This doc is really useful. I tried to create a new app, but unfortunately the command fails with the following error

```
$ npx @backstage/cli create-app
Cannot find module 'tar'
Require stack:
- XXX/.npm/_npx/12381/lib/node_modules/@backstage/cli/dist/commands/build-cache/archive.js
- XXX/.npm/_npx/12381/lib/node_modules/@backstage/cli/dist/commands/build-cache/cache.js
- XXX/.npm/_npx/12381/lib/node_modules/@backstage/cli/dist/commands/build-cache/index.js
- XXX/.npm/_npx/12381/lib/node_modules/@backstage/cli/dist/index.js
- XXX/.npm/_npx/12381/lib/node_modules/@backstage/cli/bin/backstage-cli
```

### Versions

```
$ node -v
v12.12.0
$ npm -v
6.14.4
$ npx -v
10.2.2

```

**Comment by Rugvip** [comment]
@guillaumebreton Thanks you for reporting the issue!

It should be fixed by #474, although we'll need a new version released too

**Comment by guillaumebreton** [comment]
Thank you @Rugvip I'll wait for the new release 👍

**Review by stefanalund — APPROVED** [review]

**Pull request merged**